TrafficCapture/captureKafkaOffloader/src/main/java/org/opensearch/migrations/trafficcapture/kafkaoffloader/KafkaCaptureFactory.java

```diff
@@ -65,13 +74,18 @@ public IChannelConnectionCaptureSerializer createOffloader(String connectionId)
                     // Used to essentially wrap Future returned by Producer to CompletableFuture
                     CompletableFuture cf = new CompletableFuture<>();
                     log.debug("Sending Kafka producer record: {} for topic: {}", recordId, topicNameForTraffic);
+                    MDC.put("diagnosticId", recordId);
+                    metricsLogger.info("Sending message to Kafka");
+                    MDC.remove("diagnosticId");
                     // Async request to Kafka cluster
                     producer.send(record, handleProducerRecordSent(cf, recordId));
                     // Note that ordering is not guaranteed to be preserved here
                     // A more desirable way to cut off our tree of cf aggregation should be investigated
                     singleAggregateCfRef[0] = singleAggregateCfRef[0].isDone() ? cf : CompletableFuture.allOf(singleAggregateCfRef[0], cf);
                     return singleAggregateCfRef[0];
                 } catch (Exception e) {
+                    metricsLogger.warn("[connectionId: {0}, topic: {1}] Sending message to Kafka", connectionId, topicNameForTraffic);
+                    metricsLogger.warn("Error: ", e);
```

Comment:
Can you codify what the log levels are going to be for?
My thought is that this should be ERROR.  The info level seems appropriate for successes.  I would wrap these decisions in a helper class though - see the comments at the top of the file.

Secondly, wouldn't this be better as one message?  I would format the message with `atError().setCause(e).setMessage(()->"...")` (personally, I find that positional printf style format statements are more error prone & prefer the string concatenation approach, but you could use String.format() or .setArgument() on the LoggingEventBuilder object returned by atError())